analyzers/src/SonarAnalyzer.VisualBasic/Rules/Naming/TypeParameterNameBase.cs

```diff
@@ -0,0 +1,48 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2023 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+namespace SonarAnalyzer.Rules.VisualBasic
+{
+    // This base class is only there to avoid duplication between the implementation of S119 and S2373
+    public abstract class TypeParameterNameBase : ParametrizedDiagnosticAnalyzer
```

Comment:
I think it should be doable to register both rules in 1 class, it should be less code. That way you would report both at the same place and when removing the rule it would be only 2 lines of removal.
`ConditionEvaluatesToConstant` does something similar. In the UTs you could use `WithOnlyDiagnostics`.